Tests/NIOExtrasTests/LengthFieldPrependerTest.swift

```diff
@@ -0,0 +1,413 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import NIO
+import NIOExtras
+
+private let standardDataString = "abcde"
+private let standardDataStringCount = standardDataString.utf8.count
+
+class LengthFieldPrependerTest: XCTestCase {
+    
+    private var channel: EmbeddedChannel!
+    private var encoderUnderTest: LengthFieldPrepender!
+    
+    override func setUp() {
+        self.channel = EmbeddedChannel()
+    }
+
+    func testEncodeWithUInt8HeaderWithData() throws {
+        
+        self.encoderUnderTest = LengthFieldPrepender(lengthFieldLength: .one,
+                                                     lengthFieldEndianness: .little)
+
+        try? self.channel.pipeline.add(handler: self.encoderUnderTest).wait()
```

Comment:
nit, this should be

    XCTAssertNoThrow(try self.channel.pipeline.add(.....).wait()